pkg/component/kubeapiserver/deployment.go

```diff
@@ -233,6 +233,10 @@ func (k *kubeAPIServer) reconcileDeployment(
 						Args:                     k.computeKubeAPIServerArgs(),
 						TerminationMessagePath:   corev1.TerminationMessagePathDefault,
 						TerminationMessagePolicy: corev1.TerminationMessageReadFile,
+						SecurityContext: &corev1.SecurityContext{
+							RunAsNonRoot: pointer.Bool(true),
+							RunAsUser:    pointer.Int64(65532),
```

Comment:
If the upstream images can run with this `65532` why they run as root? I think we should open upstream issue to ask also the K8s folks to change the default user for the container image to `65532`.

Reply:
They can run when set to run as `65532`, however there can be a problem with the volumes and which files are configured to be used from them.

Currently all our secret and configMap volumes have set `defaultMode: 420` which gives `644` permission to these files. Since these files are owned by `root` setting `640` would remove any permissions to the `65532` user.

I have now moved the settings to the podSecurityContext, where I can set `fsGroup` to `65532` which would change the group owner of these files to `65532` and we can lower the permissions of mounted files to `640`. There are `*.key` files which we would ideally want to have `600` permission, but currently it is not possible to change the user owner of files [ref](https://github.com/kubernetes/kubernetes/issues/81089)

In conclusion, I do not think it is needed to create such issue, since the change would be breaking and may users may have opted to use `defaultMode: 384` which would give `600` permissions.